flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderRetrievalDriver.java

```diff
@@ -114,10 +126,21 @@ public void onError(List<KubernetesConfigMap> configMaps) {
         }
 
         @Override
-        public void handleFatalError(Throwable throwable) {
-            fatalErrorHandler.onFatalError(
-                    new LeaderRetrievalException(
-                            "Error while watching the ConfigMap " + configMapName));
+        public void handleError(Throwable throwable) {
+            if (throwable instanceof KubernetesTooOldResourceVersionException) {
+                synchronized (watchLock) {
+                    if (running) {
+                        LOG.info("Creating a new watch on ConfigMap {}.", configMapName);
+                        kubernetesWatch =
```

Comment:
Is it somewhere specified that `kubernetesWatch` closes automatically if an exception occurs? If not, then I think we should still call `kubernetesWatch.close` on the old instance. It might not be required right now because of some implementation details of the existing implementations, but this might change in the future.